dill/_dill.py

```diff
@@ -2154,7 +1902,7 @@ def save_function(pickler, obj):
                 found, _ = _getattribute(module, obj.__qualname__)
                 if getattr(found, '__func__', None) is obj:
                     _pypy_builtin = True
-            except:
+            except AttributeError:
```

Comment:
Not sure if AttributeError is only possibility here...

Reply:
Let's see:
- `obj.__qualname__` may rise an `AttributeError` (AE);
- `_getattribute()` only explicitly raises AE;
- `module` may be a module or `None` (from `_import_module(safe=True)`), and in this case it will generate an AE inside `_getattribute()`;
- if `obj.__qualname__` is not a string (which would by wild, as you cannot set `__qualname__` to anything but a string), it will rise an AE if it doesn't have a `split()` method;
- only if `obj.__qualname__` is of another type that also has a `split()` method, like a `bytes` object, it will rise a `TypeError` if the method expects an argument of a different type (also `bytes` for a `bytes` object).

I would argue that in this last case raising an exception is appropriate.